aptos/token_bridge/sources/bridge_implementation.move

```diff
@@ -0,0 +1,138 @@
+            payload_id,
+            token_address,
+            token_chain,
+            decimals,
+            symbol,
+            name
+        );
+
+        let payload:vector<u8> = encode_asset_meta(_asset_meta);
+        let nonce = 0;
+        let message_fee = wormhole::state::get_message_fee();
+        let fee_coins = withdraw(user, message_fee);
+        bridge_state::publish_message(
+            nonce,
+            payload,
+            fee_coins
+        )
+    }
+
+    public entry fun create_wrapped<CoinType>(vaa: vector<u8>): address {
+        let vaa = parse_and_verify(vaa);
+        let _token_bridge_signer = token_bridge_signer();
+        let seed = b"blah"; //TODO - how to set seed?
+        let (new_signer, new_cap) = create_resource_account(&_token_bridge_signer, seed);
+        let token_address = address_of(&new_signer);
+        deploy_coin(&new_signer);
+        bridge_state::set_wrapped_asset_signer_capability(to_bytes(&token_address), new_cap);
+
+        let _asset_meta: AssetMeta = bridge_structs::parse_asset_meta(vaa::get_payload(&vaa));
+        // TODO - how to initialize a new Coin without the type as input? e.g. create_wrapped<CoinType>...
+        // pre-deploy, or figure out what the asset address will be somehow?
+        let name = bridge_structs::get_name(&_asset_meta);
+        let symbol = bridge_structs::get_symbol(&_asset_meta);
+        let decimals = bridge_structs::get_decimals(&_asset_meta);
+        let monitor_supply = false;
+        let (burn_cap, freeze_cap, mint_cap) = initialize<CoinType>(&new_signer, utf8(name), utf8(symbol), decimals, monitor_supply);
+        vaa::destroy(vaa);
+
+        //TODO - store these caps inside of state. For now we just destroy them
+        // so the code compiles
+        coin::destroy_burn_cap(burn_cap);
+        coin::destroy_freeze_cap(freeze_cap);
+        coin::destroy_mint_cap(mint_cap);
+
+        token_address
+    }
+
+    public entry fun transfer_tokens_with_payload (
+        _token: vector<u8>,
+        _amount: U256,
+        _recipientChain: U16,
+        _recipient: vector<u8>,
+        _nonce: U32,
+        _payload: vector<u8>
+    ) {
+        //TODO
+    }
+
+    /*
+     *  @notice Initiate a transfer
+     */
+    fun transfer_tokens_(_token: TypeInfo, _amount: u128, _arbiterFee: u128) {//returns TransferResult
+        // TODO
+    }
+
+    fun bridge_out(_token: vector<u8>, _normalized_amount: U256) {
+        // TODO
+        //let outstanding = outstanding_bridged(token);
+        //let lhs = u256::add(outstanding, normalized_amount);
+        //assert!(u256::compare(lhs, &(2<<128-1))==1, 0); //LHS is less than RHS
+        //setOutstandingBridged(token, u256::add(outstanding, normalized_amount));
+    }
+
+    fun bridged_in(token: vector<u8>, normalized_amount: U256) {
+        set_outstanding_bridged(token, u256::sub(outstanding_bridged(token), normalized_amount));
+    }
+
+    fun verify_bridge_vm(vm: &VAA): bool{
```

Comment:
I think this should not return a bool, but do an assertion instead, and we should call it
`assert_known_emitter`